roles/mesos/defaults/main.yml

```diff
@@ -25,8 +25,8 @@ mesos_attributes:
 mesos_cluster: mantl
 
 mesos_zk_auth: "{% if zk_mesos_user_secret is defined %}{{ zk_mesos_user }}:{{ zk_mesos_user_secret }}@{% endif %}"
-mesos_zk_dns: "zookeeper.service.consul"
-mesos_zk_hosts: "{% for host in groups[zookeeper_server_group] %}{{ host }}:{{mesos_zk_port}}{% if not loop.last %},{% endif %}{% endfor %}"
+mesos_zk_dns: "zoiokeeper.service.consul"
```

Comment:
This actually is correct:

for host in groups[zookeeper_server_group] mandates existence of the role=control key, which in the case of adding a single worker node to an existing cluster, it fails.

The code I added checks for existence of the zookeeper_hosts_override variable. If the variable does exist the groups[zookeeper_server_group] is assumed to be present, otherwise it executes an iteration through an array of overriding variables.